### PR TITLE
[16.0][MIG] crm_timesheet: Migration to version 16.0

### PR DESCRIPTION
**Pull request by carolinafernandez-tecnativa**
- Standard Migration V16
- Use the upper base view instead and remove this dependency to hr_timesheet_task_stage
- Modify test to be compatible with v16

@Tecnativa

 TT45227

@pedrobaeza

### CODE REVIEW

File: crm_timesheet/__manifest__.py [hunk 1]
```diff
@@ -3,13 +3,14 @@
...
+# Copyright 2023 Tecnativa - Carolina Fernandez
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
 {
     "name": "CRM Timesheet",
     "category": "Customer Relationship Management",
-    "version": "15.0.1.0.0",
-    "depends": ["crm", "project_timesheet_time_control"],
+    "version": "16.0.1.0.0",
+    "depends": ["crm", "project_timesheet_time_control", "hr_timesheet_task_stage"],
```
Comment (pedrobaeza) [inline crm_timesheet/__manifest__.py]: Why do you need `hr_timesheet_task_stage`?

File: crm_timesheet/views/crm_lead_view.xml [hunk 2]
```diff
@@ -28,18 +25,26 @@
...
             <page name="extra" position="after">
-                <page string="Timesheet">
+                <page string="Timesheet" groups="hr_timesheet.group_hr_timesheet_user">
+                    <field name="timesheet_ids" invisible="1" />
                     <field
                         name="timesheet_ids"
+                        groups="hr_timesheet.group_hr_timesheet_user"
                         context="{'default_project_id': project_id, 'default_user_id': uid, 'tree_view_ref': 'hr_timesheet.timesheet_view_tree_user', 'form_view_ref': 'hr_timesheet.hr_timesheet_line_form'}"
                     />
+
```
Comment (pedrobaeza) [inline crm_timesheet/views/crm_lead_view.xml]: Remove this extra empty line

**Review by pedrobaeza — APPROVED** [review]

File: crm_timesheet/__manifest__.py (hunk 1 above)
Comment (carolinafernandez-tecnativa) [reply crm_timesheet/__manifest__.py:13]: There is an inherit view in views/hr_timesheet_view.xml refering to this module: hr_timesheet_task_stage

File: crm_timesheet/views/hr_timesheet_view.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+                />
+            </field>
+        </field>
+    </record>
+    <record id="hr_timesheet_line_form" model="ir.ui.view">
+        <field name="name">Lead/opportunity in timesheet line form</field>
+        <field name="model">account.analytic.line</field>
+        <field
+            name="inherit_id"
+            ref="hr_timesheet_task_stage.account_analytic_line_form"
```
Comment (pedrobaeza) [inline crm_timesheet/views/hr_timesheet_view.xml]: Use the upper base view instead and remove this dependency.

**Review by pedrobaeza — APPROVED** [review]

**Review by victoralmau — APPROVED** [review]

**Pull request merged**